File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/EmployeeLeaveProfileForm.ts
import { DecimalEditor, initFormType, LookupEditor, PrefixedContext } from "@serenity-is/corelib";

export interface EmployeeLeaveProfileForm {
    EmployeeId: LookupEditor;
    LeavePolicyId: LookupEditor;
    LeaveTypeId: LookupEditor;
    OpeningBalance: DecimalEditor;
    AccruedLeave: DecimalEditor;
    UsedLeave: DecimalEditor;
    PendingLeave: DecimalEditor;
    CarryForwardLeave: DecimalEditor;
    LOPDays: DecimalEditor;
}

export class EmployeeLeaveProfileForm extends PrefixedContext {
    static readonly formKey = 'Operations.EmployeeLeaveProfile';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeLeaveProfileForm.init) {
            EmployeeLeaveProfileForm.init = true;

            var w0 = LookupEditor;
            var w1 = DecimalEditor;

            initFormType(EmployeeLeaveProfileForm, [
                'EmployeeId', w0,
                'LeavePolicyId', w0,
                'LeaveTypeId', w0,
                'OpeningBalance', w1,
                'AccruedLeave', w1,
                'UsedLeave', w1,
                'PendingLeave', w1,
                'CarryForwardLeave', w1,
                'LOPDays', w1
            ]);
        }
    }
}

File: HRMS/HRMS.Web/Modules/Operations/EmployeeLeaveProfile/EmployeeLeaveProfileForm.cs
using Serenity.ComponentModel;

namespace HRMS.Operations.Forms;

[FormScript("Operations.EmployeeLeaveProfile")]
[BasedOnRow(typeof(EmployeeLeaveProfileRow), CheckNames = true)]
public class EmployeeLeaveProfileForm
{
    public int EmployeeId { get; set; }
    public int LeavePolicyId { get; set; }
    public int LeaveTypeId { get; set; }
    public decimal OpeningBalance { get; set; }
    public decimal AccruedLeave { get; set; }
    public decimal UsedLeave { get; set; }
    public decimal PendingLeave { get; set; }
    public decimal CarryForwardLeave { get; set; }
    public decimal LOPDays { get; set; }
}
